py_neuromodulation/nm_stream_offline.py

```diff
@@ -133,23 +138,35 @@ def _run(
                 data_batch.astype(np.float64)
             )
             if is_stream_lsl:
-                feature_dict["time"] = time_[-1]
                 if self.verbose:
                     if last_time is not None:
                         logger.debug("%.3f seconds of new data processed", time_[-1] - last_time)
+                        feature_dict["time"] = time_[-1] - last_time
+                    else:
+                        feature_dict["time"] = 0
                     last_time = time_[-1]
             else:
-                feature_dict["time"] = np.ceil(time_[-1] * 1000 +1 ).astype(int)
+                feature_dict["time"] = np.ceil(time_[-1] * 1000 +1 )
                 logger.info("Time: %.2f", feature_dict["time"]/1000)
             
 
             self._add_target(feature_dict, data_batch)
+            buff_cnt += 1
+
+            feature_dict = db.cast_values(feature_dict)
+            db.create_table(feature_dict)
```

Comment:
Move up above while loop

Reply:
It has to stay in the while loop to init the table with the values from the loop. 
Solved this by putting it in an if statement